AndroidSDK/src/com/leanplum/internal/Request.java

```diff
@@ -383,11 +384,13 @@ private void sendNow() {
     Util.executeAsyncTask(true, new AsyncTask<Void, Void, Void>() {
       @Override
       protected Void doInBackground(Void... params) {
-        final List<Map<String, Object>> requestsToSend = getUnsentRequests();
+        List<Map<String, Object>> requestsToSend = getUnsentRequests();
         if (requestsToSend.isEmpty()) {
           return null;
         }
 
+        requestsToSend = updateUnsentRequests(requestsToSend);
```

Comment:
why do we need to update unsent requests ? couldn't we generate uuid when inserting ? This just causes double reading/parsing and saving again.